src/abundances.jl

```diff
@@ -1,62 +1,59 @@
 # Methods for absolute and relative abundances
 
-struct AbundanceTable{T<:Real} <: AbstractArray{T,2}
-    table::Array{T,2}
-    samples::Vector{S} where S
-    features::Vector{R} where R
-end
-
-function AbundanceTable(df::DataFrame)
-    return AbundanceTable(
-            Matrix(df[:,2:end]),
-            names(df[2:end]),
-            Vector(df[1]))
-end
-function AbundanceTable(table::Array{T,2}) where T<:Real
-    return AbundanceTable(
-             table,
-             Vector{Int64}(1:size(table,2)),
-             Vector{Int64}(1:size(table,1)))
-end
-
-@forward_func AbundanceTable.table Base.getindex, Base.setindex, Base.length, Base.size
+abundancetable(df::DataFrame) = ComMatrix(df, sitecolumns = true)
```

Comment:
I've made some constructors that take data in the colums-are-sites format you're using, and then transpose it before passing to ComMatrix. I'm happy to discuss which is the best design on rows and columns and then if necessary change it in SpatialEcology, but we may need more opinions on this. Hopefully the interface (using `species` etc) means that the internal transpose won't make it harder to program to.

Reply:
Yeah, I don't quite know how to think about this yet. Happy to try it this way for a bit and see what it breaks...

Reply:
No doubt this is the main sticking point